library/src/main/scala/scala/scalajs/js/Promise.scala

```diff
@@ -37,26 +37,35 @@ import scala.concurrent.Future
  *  directly use the methods of `Future` on `Promise`s.
  */
 @js.native
-class Promise[+A](
+class Promise[+A] private[js] (
```

Comment:
You can't hide the constructor. It must be accessible for subclasses. (because yes, `Promise` is subclassable in ES)